### PR TITLE
CachedOpEnvironment: use hash object instead of long

### PR DESCRIPTION
**Pull request by dietzc**
Avoids auto-boxing when passed to cache, which confuses several cache
implementation (e.g. guava). Guava cache didn't work with passed primitive `long`.

@ctrueden you think thats OK?

### CODE REVIEW

File: src/main/java/net/imagej/ops/cached/CachedOpEnvironment.java [hunk 1]
```diff
@@ -244,10 +228,42 @@ public O createOutput(I input) {
...
-		
+
 		@Override
 		public CachedHybridOp<I, O> getIndependentInstance() {
 			return this;
 		}
 	}
+
+	/**
+	 * Simple helper class to generate an object which
```
Comment (ctrueden) [inline src/main/java/net/imagej/ops/cached/CachedOpEnvironment.java]: Unfinished sentence.

File: src/main/java/net/imagej/ops/cached/CachedOpEnvironment.java [hunk 2]
```diff
@@ -53,9 +53,8 @@
 
 	@Parameter
 	private CacheService cs;
-	
-	public CachedOpEnvironment(final OpEnvironment parent)
-	{
+
+	public CachedOpEnvironment(final OpEnvironment parent) {
```
Comment (ctrueden) [inline src/main/java/net/imagej/ops/cached/CachedOpEnvironment.java]: Style changes are mixed in.

File: src/main/java/net/imagej/ops/cached/CachedOpEnvironment.java (hunk 1 above)
Comment (dietzc) [inline src/main/java/net/imagej/ops/cached/CachedOpEnvironment.java]: oh wow... I will fix that

**Comment by ctrueden** [comment]
Seems OK to me!

**Comment by dietzc** [comment]
I will get rid of the style changes (thank you mars...grrr...) and merge afterwards. thanks!

**Comment by dietzc** [comment]
@ctrueden if you like it, you can merge it. Thanks!

**Pull request merged**